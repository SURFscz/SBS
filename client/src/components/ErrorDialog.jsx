import React from "react";
import Modal from "react-modal";
import "./ConfirmationDialog.scss";
import I18n from "i18n-js";
import {stopEvent} from "../utils/Utils";

export default function ErrorDialog({isOpen = false, close}) {
    return (
        <Modal
            isOpen={isOpen}
            onRequestClose={close}
            contentLabel={I18n.t("error_dialog.title")}
            className="confirmation-dialog-content"
            overlayClassName="confirmation-dialog-overlay"
            closeTimeoutMS={250}
            ariaHideApp={false}>
            <section className="dialog-header error">
                {I18n.t("error_dialog.title")}
            </section>
            <section className="dialog-content">
                <h2>{I18n.t("error_dialog.body")}</h2>
            </section>
            <section className="dialog-buttons">
                <a href="/close" className="button white error" onClick={e => {
                    stopEvent(e);
                    close(e);
                }}>{I18n.t("error_dialog.ok")}</a>
            </section>
        </Modal>
    );

}

