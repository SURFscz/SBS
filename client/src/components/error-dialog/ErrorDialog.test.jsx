// client/src/components/error-dialog/ErrorDialog.test.jsx
import React from "react";
import { render } from "@testing-library/react";
import ErrorDialog from "./ErrorDialog";

describe("ErrorDialog", () => {
  it("matches snapshot when open", () => {
    const { asFragment } = render(<ErrorDialog isOpen={true} close={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders nothing when closed", () => {
    const { container } = render(<ErrorDialog isOpen={false} close={() => {}} />);
    expect(container.firstChild).toBeNull();
  });
});
