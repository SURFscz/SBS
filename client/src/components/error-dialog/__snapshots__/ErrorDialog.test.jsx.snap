// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDialog matches snapshot when open 1`] = `
<DocumentFragment>
  <div
    class="sds--modal sds--backdrop"
  >
    <div
      class="sds--modal--container  welcome-dialog"
    >
      <div
        class="sds--modal--title "
      >
        <p
          class="sds--text--h4"
        >
          Unexpected error
        </p>
      </div>
      <div
        class="sds--alert sds--alert--status-error  "
      >
        <div
          class="sds--alert--inner "
        >
          <div
            class="sds--alert--visual"
          >
            <svg
              height="22.6"
              viewBox="0 0 22.6 22.6"
              width="22.6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="bg"
                d="M12.8 1.8c-.1-.3-.4-.6-.6-.7q-.45-.3-.9-.3-.6 0-.9.3c-.3.3-.5.3-.7.6L.9 19.9c-.1.2-.2.5-.2.7 0 .3.1.5.2.7s.3.4.6.5c.2.1.5.2.7.2h18.3c.3 0 .5-.1.7-.2s.4-.3.6-.5c.1-.2.2-.5.2-.7 0-.3 0-.5-.2-.7z"
                fill="none"
              />
              <path
                d="M12.8 1.8c-.1-.3-.4-.6-.6-.7q-.45-.3-.9-.3-.6 0-.9.3c-.3.3-.5.3-.7.6L.9 19.9c-.1.2-.2.5-.2.7 0 .3.1.5.2.7s.3.4.6.5c.2.1.5.2.7.2h18.3c.3 0 .5-.1.7-.2s.4-.3.6-.5c.1-.2.2-.5.2-.7 0-.3 0-.5-.2-.7z"
                fill="none"
                stroke="#000"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M11.3 17.7c-.1 0-.1 0-.2.1-.1 0-.1.1-.1.2v.2c0 .1.1.1.1.2.1.1.1.1.2.1h.2c.1 0 .1-.1.2-.1 0-.1.1-.1.1-.2s0-.2-.1-.3c-.2-.2-.3-.2-.4-.2"
                fill="none"
                stroke="#000"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M11.3 14.9V7.8"
                fill="none"
                stroke="#000"
                stroke-linecap="round"
                stroke-miterlimit="10"
                stroke-width="1.5"
              />
            </svg>
          </div>
          <div
            class="sds--alert--textual"
          >
            <p>
              An error has occurred
            </p>
          </div>
        </div>
      </div>
      <div
        class="sds--modal--content"
      >
        <div
          class="sds--text--rich"
        >
          <section
            class="dialog-content"
          >
            <p>
              This is embarrassing; an unexpected error has occurred. It has been logged and reported. Please try again...
            </p>
          </section>
        </div>
        <div
          class="sds--modal--actions sds--actions--outer sds--space--top--5"
        >
          <button
            class="sds--btn sds--btn--primary    "
            type="button"
          >
            <span
              class="textual"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
